### ISSUE TITLE
Flags --gasprice and --txpool.pricelimit seem to define the same attribute

### ISSUE THREAD

**Issue by markya0616** [ethereum/go-ethereum#14715]
#### System information

Geth version: `1.6.6`

#### Expected behaviour
We set `--gasprice` to 0, and expect that we can accept transactions with 0 gasprice.

#### Actual behaviour
Fail to accept transactions with 0 gasprice if we set `--gasprice` to 0, but success if we set `--txpool.pricelimit` to 0.

```
--gasprice "18000000000"  Minimal gas price to accept for mining a transactions
--txpool.pricelimit value    Minimum gas price limit to enforce for acceptance into the pool (default: 1)
```

Do these two flags define the same attribute? If so, I can fire a PR to combine these two flags.

**Comment by karalabe** [comment]
They are not exactly the same. The `txpool.pricelimit` defines a baseline transaction price under which the node will simply not accept transactions (not even for forwarding). The `gasprice` flag is only used during mining, but if you do enable mining, it overwrites the former price limit to enforce that only "mineable" transactions end up in your pool.

**Comment by markya0616** [comment]
Thanks for the feedback. However, in my observation, if we start geth with flag `--gasprice` and `--mine`, the `gasprice` cannot overwrite the `txpool.pricelimit`. Is it an issue?

**Comment by karalabe** [comment]
That was a bug, which should be fixed in 1.6.6. Are you sure you're still seeing this issue?

**Comment by markya0616** [comment]
This issue is fixed after I pull the latest code. Thanks for the feedbacks!
